h5py/utils.pyx

```diff
@@ -164,7 +164,10 @@ cdef object create_hsize_array(object arr):
     else:
         raise RuntimeError("Can't map hsize_t %d to Numpy typecode" % sizeof(hsize_t))
 
-    return PyArray_FROM_OTF(arr, typecode, NPY_CONTIGUOUS | NPY_NOTSWAPPED | NPY_FORCECAST)
+    return PyArray_FROM_OTF(arr, typecode,
+        NPY_ARRAY_C_CONTIGUOUS
```

Comment:
This replacement wasn't immediately clear to me: there's no `NPY_ARRAY_CONTIGUOUS`, but inspecting numpy's source makes it clear why: `NPY_CONTIGUOUS` was always an alias for `NPY_C_CONTIGUOUS`
see https://github.com/numpy/numpy/blob/fd8a68e4978defd094cffa23c71d7de7fb213e3f/numpy/_core/include/numpy/npy_1_7_deprecated_api.h#L31